test/js/node/tls/fetch-tls-cert.test.ts

```diff
@@ -0,0 +1,360 @@
+import { expect, it } from "bun:test";
+import tls from "tls";
+import type { Server, TLSSocket } from "node:tls";
+import type { AddressInfo } from "node:net";
+import { join } from "path";
+import { readFileSync } from "fs";
+
+const client = {
+  key: readFileSync(join(import.meta.dir, "fixtures", "ec10-key.pem"), "utf8"),
+  cert: readFileSync(join(import.meta.dir, "fixtures", "ec10-cert.pem"), "utf8"),
+  ca: readFileSync(join(import.meta.dir, "fixtures", "ca5-cert.pem"), "utf8"),
+};
+const server = {
+  key: readFileSync(join(import.meta.dir, "fixtures", "agent10-key.pem"), "utf8"),
+  cert: readFileSync(join(import.meta.dir, "fixtures", "agent10-cert.pem"), "utf8"),
+  ca: readFileSync(join(import.meta.dir, "fixtures", "ca2-cert.pem"), "utf8"),
+};
+
+function split(file: any, into: any) {
+  const certs = /([^]*END CERTIFICATE-----\r?\n)(-----BEGIN[^]*)/.exec(file) as RegExpExecArray;
+  into.single = certs[1];
+  into.subca = certs[2];
+}
+
+// Split out the single end-entity cert and the subordinate CA for later use.
+split(client.cert, client);
+split(server.cert, server);
+
+// The certificates aren't for "localhost", so override the identity check.
+function checkServerIdentity(hostname: string, cert: any) {
+  expect(hostname).toBe("localhost");
+  expect(cert.subject.CN).toBe("agent10.example.com");
+}
+
+async function connect(options: any) {
+  let server: ReturnType<Bun.serve> | null = null;
+  try {
+    server = Bun.serve({
+      tls: options.server,
+      port: 0,
+      fetch(req) {
+        return new Response("Hello World!");
+      },
+    });
+    const port = server.port;
+    const result = await fetch(`https://localhost:${port}`, {
+      tls: options.client,
+    }).then(res => res.text());
+    if (result !== "Hello World!") {
+      throw new Error("Unexpected response from server");
+    }
+  } finally {
+    server?.stop(true);
+  }
+}
+it.todo("complete cert chains sent to peer.", async () => {
+  await connect({
+    client: {
+      key: client.key,
+      cert: client.cert,
+      ca: server.ca,
+      checkServerIdentity,
+    },
+    server: {
+      key: server.key,
+      cert: server.cert,
+      ca: client.ca,
+      requestCert: true,
+    },
+  });
+});
+
+it.todo("complete cert chains sent to peer, but without requesting client's cert.", async () => {
```

Comment:
a lot of this is `todo` because we still need to add these options to `Bun.serve` and/or `fetch`